### PR TITLE
 Add input validation for prelu.py and leaky_relu.py (Fixes #22162)

### LINKED ISSUE #22162: Missing input validation for PReLU alpha_initializer and LeakyReLU alpha
### Summary
`keras.layers.PReLU` and `keras.layers.LeakyReLU` lack sufficient input validation during instantiation and the `build()` phase. 

1. **PReLU:** Accepts `alpha_initializer=None`, leading to an inconsistent state that results in downstream failures during training or serialization.
2. **LeakyReLU:** Accepts `NaN` values for the `alpha` parameter, allowing the creation of a broken layer that produces non-numeric gradients.

This is a **usability and reliability bug** that violates the "Fail-Fast" principle of the Keras API.

### Reproduction Script
```python
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
import keras
import numpy as np

print(f"Testing Keras version: {keras.__version__}")

# --- Case 1: PReLU ---
print("\n--- Testing PReLU ---")
try:
    layer_prelu = keras.layers.PReLU(alpha_initializer=None)
    layer_prelu.build((None, 10)) 
    print("[ISSUE] PReLU accepted None as alpha_initializer and built successfully!")
except Exception as e:
    print(f"[OK] PReLU caught error: {e}")

# --- Case 2: LeakyReLU ---
print("\n--- Testing LeakyReLU ---")
try:
    layer_leaky = keras.layers.LeakyReLU(alpha=float('nan'))
    print("[ISSUE] LeakyReLU accepted NaN as alpha!")
except Exception as e:
    print(f"[OK] LeakyReLU caught NaN.")
```

### Observed Result
```text
Testing Keras version: 3.14.0
--- Testing PReLU ---
[ISSUE] PReLU accepted None as alpha_initializer and built successfully!

--- Testing LeakyReLU ---
[ISSUE] LeakyReLU accepted NaN as alpha!
```

### Actual Behavior
The layers are instantiated and built without any validation errors. The `PReLU` layer remains in an inconsistent state, and `LeakyReLU` accepts mathematically undefined values (`NaN`), deferring crashes to the training phase.

### Expected Behavior
The constructor should raise a `ValueError` immediately if:
- `alpha_initializer` is `None` in `PReLU`.
- The `alpha` (or `negative_slope`) parameter in `LeakyReLU` is `NaN`.

### Root Cause
In `keras/src/layers/activations/prelu.py` and `leaky_relu.py`, parameters are assigned to the object's state without being passed through a validation utility. This creates a "silent failure" scenario where architecture errors are caught too late in the ML lifecycle.

### Suggested Fix

**For PReLU:**
```python
if alpha_initializer is None:
    raise ValueError("The `alpha_initializer` argument for PReLU cannot be None.")
```

**For LeakyReLU:**
```python
if np.isnan(alpha):
    raise ValueError("The `alpha/negative_slope` argument for LeakyReLU cannot be NaN.")
```

### Environment
* **OS:** Windows 11
* **Python:** 3.13.5
* **Keras:** 3.14.0
* **Backend:** TensorFlow 2.20.0

### Conclusion
Implementing strict input validation ensures that architecture inconsistencies are caught during the model definition phase, preventing silent failures and wasted computational resources during training execution.

### PR DESCRIPTION
**Pull request by adamsaidane**
## Description
This PR adds input validation to the PReLU and LeakyReLU activation layers to catch invalid parameters during layer instantiation rather than deferring errors to the training phase.

## Changes Made
- **PReLU**: Added validation to reject `None` as `alpha_initializer`
- **LeakyReLU**: Added validation to reject `NaN` as `negative_slope`
- Added comprehensive test cases for both validation checks

## Motivation
As described in #22162, these layers currently accept invalid parameters without raising errors, leading to:
- Inconsistent layer states (PReLU with None initializer)
- Mathematical errors during training (LeakyReLU with NaN)
- Wasted computational resources

This fix implements the "fail-fast" principle, ensuring errors are caught at model definition time.

## Testing
- Added `test_prelu_alpha_initializer_is_none()`
- Added `test_negative_slope_nan_value()`
- Verified all existing tests still pass
- Tested edge cases with the reproduction script from #22162

Fixes #22162

### CODE REVIEW

**Comment by adamsaidane** [comment]
I've investigated the failing ReLUTest::test_invalid_usage test. This is a pre-existing OpenVINO backend issue, not caused by these PReLU/LeakyReLU validation changes.

Verification:
✅ All activation tests pass with other backends
❌ This test fails on master with OpenVINO backend (confirmed locally)
❌ Same failure appears in this PR (same root cause)

The error "expected non-negative integer" is coming from OpenVINO's backend validation layer, not from the ReLU layer itself. Since this PR only modifies PReLU and LeakyReLU validation, this pre-existing OpenVINO compatibility issue should not block the merge.

Recommend opening a separate issue to investigate ReLU validation on the OpenVINO backend.
@gbaned

**Comment by hertschuh** [comment]
Hi,

I'm going to have to close this PR as I question the premise of the issue, and for the consistency of code.

- passing a negative slope of NaN is obviously going to yield NaNs, I don't think it would be a reasonable approach to check for NaN everywhere there's a parameter in the code.
- passing None to an initializer that has a reasonable default non-None value is obviously not going to work. And if we wanted to check initializers for None values, we should do it consistently across the codebase.
